eng/tox/mypy_hard_failure_packages.py

```diff
@@ -6,5 +6,6 @@
 # --------------------------------------------------------------------------------------------
 
 MYPY_HARD_FAILURE_OPTED = [
-  "azure-core"
+  "azure-core",
+  "azure-storage-blob"
```

Comment:
Opt-in looks right to me! Can you confirm that you have seen a MyPy run fail at least once?

Reply:
Nevermind, I can see the additional log output confirming it's opt-ed in.